lib/tasks/gettext.rake

```diff
@@ -35,6 +35,31 @@ namespace :gettext do
         end
    end
 
+   desc 'Update locale files with slightly changed English msgids using a csv file of old to new strings'
+   task :update_msgids_from_csv do
+       mapping_file = find_mapping_file(ENV['MAPPING_FILE'])
+       mappings = {}
+       File.foreach(mapping_file) do |csv_line|
+           from,to = CSV.parse(csv_line.gsub('\"', '""')).first
+           mappings[from] = to
+       end
+       Dir.glob("locale/**/app.po").each do |po_file|
+           lang_mappings = mappings.clone
+           lines = []
```

Comment:
Actually I think I've misunderstood – we're writing the entire file, except swapping out the updated line on L53?